### PR TITLE
Adds feedback to the supermatter exploding while in a locker

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
You will now both hear the sound and receive a special message when the supermatter explodes while you're in a locker.

## Why It's Good For The Game
Currently, when the supermatter explodes and you are in a locker, it is silent, and you just have to time getting out. This should speed everything up.

## Changelog
:cl:
add: You will now hear the sound and receive a distinct message when the supermatter explodes while you are safe.
/:cl:

### CODE REVIEW

**Comment by zxaber** [comment]
Is lockers shielding you from the effects of the SM actually intentional? I'd always assumed it was a bug no one bothered to fix.

**Comment by Jared-Fogle** [comment]
It's a good feature, regardless of if it's intentional or not. It keeps the supermatter delaminating as a round defining event (you have to drop EVERYTHING you're doing and get in a locker, while everyone screams over the radio), while making the rest of the round not unbearable to play. The shuttle gets called no matter what, and people will still forget to protect themselves, so medbay gets the flood of having to fix up a ton of irradiated people in the meantime.

**Comment by CoffeeDragon16** [comment]
I always thought it was weird and awkward that there was no feedback for if you were in a locker. Occasionally if the server was slow you'd pass what should be the expected time, pop out, and the hallucinations happen anyway so it's really nice to have a confirmation that you're not gonna suddenly see greytiders in the walls

**Comment by LemonInTheDark** [comment]
probably unintended, the comments imply it should hit locker'd people hardest, but it's not a big enough deal to matter

File: code/modules/power/supermatter/supermatter.dm [hunk 1]
```diff
@@ -422,10 +422,15 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
...
 			SEND_SOUND(M, 'sound/magic/charge.ogg')
-			to_chat(M, "<span class='boldannounce'>You feel reality distort for a moment...</span>")
-			SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "delam", /datum/mood_event/delam)
+
+			if (M.z == z)
+				to_chat(M, "<span class='boldannounce'>You feel reality distort for a moment...</span>")
+				SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "delam", /datum/mood_event/delam)
+			else
+				to_chat(M, "<span class='boldannounce'>You hold onto \the [M.loc] as hard as you can, as reality distorts around you. You feel safe.</span>")
+
```
Comment (LemonInTheDark) [inline code/modules/power/supermatter/supermatter.dm:433]: src in this instance is the shard itself.  You'll want to run a get turf for each mob. This whole interaction feels a bit jank to be honest, someone wasn't thinking about lockers properly when they wrote the hallucination code.

Comment (Jared-Fogle) [reply code/modules/power/supermatter/supermatter.dm:433]: Oh, you're right. I'll fix it today.

**Pull request merged**